app/src/main/java/org/thoughtcrime/securesms/payments/MobileCoinMainNetConfig.java

```diff
@@ -0,0 +1,93 @@
+  }
+
+  @Override
+  @NonNull AuthCredentials getAuth() throws IOException {
+    return signalServiceAccountManager.getPaymentsAuthorization();
+  }
+
+  @Override
+  @NonNull ClientConfig getConfig() {
+    try {
+      byte[] mrEnclaveConsensus    = Hex.toByteArray("e66db38b8a43a33f6c1610d335a361963bb2b31e056af0dc0a895ac6c857cab9");
+      byte[] mrEnclaveConsensusNew = Hex.toByteArray("653228afd2b02a6c28f1dc3b108b1dfa457d170b32ae8ec2978f941bd1655c83");
+      byte[] mrEnclaveReport       = Hex.toByteArray("709ab90621e3a8d9eb26ed9e2830e091beceebd55fb01c5d7c31d27e83b9b0d1");
+      byte[] mrEnclaveReportNew    = Hex.toByteArray("f3f7e9a674c55fb2af543513527b6a7872de305bac171783f6716a0bf6919499");
+      byte[] mrEnclaveLedger       = Hex.toByteArray("511eab36de691ded50eb08b173304194da8b9d86bfdd7102001fe6bb279c3666");
+      byte[] mrEnclaveLedgerNew    = Hex.toByteArray("89db0d1684fcc98258295c39f4ab68f7de5917ef30f0004d9a86f29930cebbbd");
+      byte[] mrEnclaveView         = Hex.toByteArray("ddd59da874fdf3239d5edb1ef251df07a8728c9ef63057dd0b50ade5a9ddb041");
+      byte[] mrEnclaveViewNew      = Hex.toByteArray("dd84abda7f05116e21fcd1ee6361b0ec29445fff0472131eaf37bf06255b567a");
+
+      Set<X509Certificate> trustRoots          = getTrustRoots(R.raw.signal_mobilecoin_authority);
+      ClientConfig         config              = new ClientConfig();
+      String[]             hardeningAdvisories = { "INTEL-SA-00334" };
+
+      config.logAdapter = new MobileCoinLogAdapter();
+      config.fogView    = new ClientConfig.Service().withTrustRoots(trustRoots)
+                                                    .withVerifier(new Verifier().withMrEnclave(mrEnclaveView, null, hardeningAdvisories)
+                                                                                .withMrEnclave(mrEnclaveViewNew, null, hardeningAdvisories));
+      config.fogLedger  = new ClientConfig.Service().withTrustRoots(trustRoots)
+                                                    .withVerifier(new Verifier().withMrEnclave(mrEnclaveLedger, null, hardeningAdvisories)
+                                                                                .withMrEnclave(mrEnclaveLedgerNew, null, hardeningAdvisories));
+      config.consensus  = new ClientConfig.Service().withTrustRoots(trustRoots)
+                                                    .withVerifier(new Verifier().withMrEnclave(mrEnclaveConsensus, null, hardeningAdvisories)
+                                                                                .withMrEnclave(mrEnclaveConsensusNew, null, hardeningAdvisories));
+      config.report     = new ClientConfig.Service().withVerifier(new Verifier().withMrEnclave(mrEnclaveReport, null, hardeningAdvisories)
+                                                                                .withMrEnclave(mrEnclaveReportNew, null, hardeningAdvisories));
+      return config;
+    } catch (AttestationException ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)